WIP: Verification of ONT ID ownership in web environment

backslash47 opened the pull request:
I propose an method which should outline necessary steps for ONT ID ownership verification based on same principles as Transaction signing and verification in Ontology blockchain.

backslash47 commented:
I've made some algorithm changes and clarified the description. I've also added implementation and applicable test cases.

xiaomin49 commented:
Thanks Backslash47 . We have reviewed the proposal . We think signature specification should to be enhanced and unified .  Key points are as below.
1.  support PublicKeyId . As you see claim_spec shows  , we hope it is  unified. 
2.  specification  for JSON keys sorting rules .  (Claim spec will describe the details )
3.  HASH should be included in Sigature procedure. (Claim spec will describe the details )

backslash47 commented:
1. I updated the proposal to use PublicKeyId instead of embedding the private key. 
2. I specifically clarified the need to use deterministic Stringify method so the same JSON input produce same string output.
3. I removed the concrete usage of SHA256 and instead mention only hashing method as part of signature schema.
4. I also modified the typescript implementation and test cases.

If you think something else is needed to be addressed, please tell.
